### DISCUSSION TITLE
Feature: Convert quotes to curly/typographical quotes plugin

### DISCUSSION THREAD

**Discussion by jonasb** [facebook/lexical#3394] [Q&A]
When writing quotes (both double and single) in the editor I'd like them to be converted to from straight to curly quotes. E.g. "it's" ⇒ “it’s”.

It seems like a common feature, so requesting it here to see if anyone already have an implementation and check if it ought to be in Playground or even a standard plugin.

Since the type of quote normally changes in editors on the context (e.g. opening double quote before a word, and closing double quote after a word), should the type of character only be defined when typing the quote, or should it change when editing the text?

There might be contexts, e.g. a code block, where you don't want curly quotes, how can that be achieved in a common plugin?

**Comment by moy2010** [comment]
Throwing some random ideas about how you could implement some of the functionality you are looking for.

For matching the action of opening and closing quotes, you could get some inspiration from the logic for text match transformers in the `lexical-markdown` package. 

> There might be contexts, e.g. a code block, where you don't want curly quotes, how can that be achieved in a common plugin?

You can check if the text node is in such a context by checking the text node's parent or top level element. Either of those should tell you if any ancestor is a code block.
